eodag/api/core.py

```diff
@@ -583,7 +647,9 @@ def list_product_types(
         :returns: The list of the product types that can be accessed using eodag.
         :raises: :class:`~eodag.utils.exceptions.UnsupportedProvider`
         """
-        if fetch_providers:
+
+        strict_mode = is_env_var_true("EODAG_STRICT_PRODUCT_TYPES")
+        if fetch_providers and not strict_mode:
```

Comment:
`fetch_product_types_list()` is called at several places in the code. Strict mode might better be checked inside `fetch_product_types_list` to avoid having to check `EODAG_STRICT_PRODUCT_TYPES` before every method call